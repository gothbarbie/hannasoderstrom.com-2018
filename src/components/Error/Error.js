// @flow

import React from 'react'
import styled from 'styled-components'
import { connect } from 'react-redux'

import imgPanda from '../../images/sadpanda.gif'

import Wrapper from '../Layout/Wrapper'
import H1 from '../Typography/H1'
import P from '../Typography/P'

type MessageType = { response?: { status?: number } }

type ErrorType = {
  error: boolean,
  message?: MessageType,
}

type ErrorProps = {
  error: ErrorType,
}

const ErrorContainer = styled.section`
  p:last-of-type {
    margin-top: 0;
  }

  h1 {
    margin: 0;
  }
`

const ErrorMessage = styled.p`
  color: white;
  font-weight: 500;
  margin: 1rem 0;
  border: 1px solid red;
  background: rgba(255, 0, 0, 0.5);
  padding: 1rem;
`

const Img = styled.img`
  width: 100%;
  height: auto;
  border-radius: 5vw;
`

export const Error = ({ error }: ErrorProps) => {
  if (!error.error || !error.message) return null

  const { message } = error

  const status = message.response && message.response.status

  let userFriendlyMsg = ''
  let userFriendlyTitle = ''

  switch (status) {
    case 404:
      userFriendlyTitle = 'Sorry!'
      userFriendlyMsg =
        'Could not retrieve content. This is most likely a temporary issue.'
      break

    default:
      break
  }

  return (
    <ErrorContainer>
      <Wrapper column>
        <H1>{userFriendlyTitle}</H1>
        {userFriendlyMsg && <ErrorMessage>{userFriendlyMsg}</ErrorMessage>}
        {userFriendlyMsg && <P>(Please enjoy this cute panda cub instead)</P>}
        {userFriendlyMsg && <Img alt="sad panda" src={imgPanda} />}
      </Wrapper>
    </ErrorContainer>
  )
}

const mapStateToProps = ({ error }) => {
  return {
    error,
  }
}

export default connect(mapStateToProps)(Error)
