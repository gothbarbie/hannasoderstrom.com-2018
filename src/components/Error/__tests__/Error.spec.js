import React from 'react'
import configureMockStore from 'redux-mock-store'
import { render } from 'enzyme'

import { Error } from '../Error'

const mockStore = configureMockStore()

describe('components/Error', () => {
  let component
  let store

  beforeEach(() => {
    store = mockStore()

    component = render(
      <Error
        error={{ error: true, message: { response: { status: 404 } } }}
        store={store}
      />
    )
  })

  it('renders', () => {
    expect(component).toMatchSnapshot()
  })
})
