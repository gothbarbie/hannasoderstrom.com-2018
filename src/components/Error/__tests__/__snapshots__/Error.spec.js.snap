// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Error renders 1`] = `
<section
  class="sc-bxivhb btGVcU"
>
  <div
    class="sc-bdVaJa fohUwm"
  >
    <h1
      class="sc-bwzfXH MAjgz"
    >
      Sorry!
    </h1>
    <p
      class="sc-ifAKCX eydVvC"
    >
      Could not retrieve content. This is most likely a temporary issue.
    </p>
    <p
      class="sc-htpNat ibeeTC"
    >
      (Please enjoy this cute panda cub instead)
    </p>
    <img
      alt="sad panda"
      class="sc-EHOje btkFze"
      src="sadpanda.gif"
    >
      
    </img>
  </div>
</section>
`;
